sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/TraceConfigzZPageHandler.java

```diff
@@ -0,0 +1,348 @@
+    out.print("<form class=\"form-flex\" action=\"" + TRACE_CONFIGZ_URL + "\" method=\"get\">");
+    out.print(
+        "<input type=\"hidden\" name=\"action\" value=\"" + QUERY_STRING_ACTION_CHANGE + "\" />");
+    emitChangeTable(out);
+    // Button for submit
+    out.print("<button class=\"button\" type=\"submit\" value=\"Submit\">Submit</button>");
+    out.print("</form>");
+    // Button for restore default
+    out.print("<form class=\"form-flex\" action=\"" + TRACE_CONFIGZ_URL + "\" method=\"get\">");
+    out.print(
+        "<input type=\"hidden\" name=\"action\" value=\"" + QUERY_STRING_ACTION_DEFAULT + "\" />");
+    out.print("<button class=\"button\" type=\"submit\" value=\"Submit\">Restore Default</button>");
+    out.print("</form>");
+    out.print("<h2>Active Tracing Parameters</h2>");
+    emitActiveTable(out);
+  }
+
+  @Override
+  public void emitHtml(Map<String, String> queryMap, OutputStream outputStream) {
+    // PrintStream for emiting HTML contents
+    try (PrintStream out = new PrintStream(outputStream, /* autoFlush= */ false, "UTF-8")) {
+      out.print("<!DOCTYPE html>");
+      out.print("<html lang=\"en\">");
+      out.print("<head>");
+      out.print("<meta charset=\"UTF-8\">");
+      out.print(
+          "<link rel=\"shortcut icon\" href=\"data:image/png;base64,"
+              + ZPageLogo.getFaviconBase64()
+              + "\" type=\"image/png\">");
+      out.print(
+          "<link href=\"https://fonts.googleapis.com/css?family=Open+Sans:300\""
+              + "rel=\"stylesheet\">");
+      out.print(
+          "<link href=\"https://fonts.googleapis.com/css?family=Roboto\" rel=\"stylesheet\">");
+      out.print("<title>" + TRACE_CONFIGZ_NAME + "</title>");
+      emitHtmlStyle(out);
+      out.print("</head>");
+      out.print("<body>");
+      try {
+        // Apply updated trace configuration based on query parameters
+        applyTraceConfig(queryMap);
+        emitHtmlBody(out);
+      } catch (Throwable t) {
+        out.print("Error while generating HTML: " + t.toString());
+        logger.log(Level.WARNING, "error while generating HTML", t);
+      }
+      out.print("</body>");
+      out.print("</html>");
+    } catch (Throwable t) {
+      logger.log(Level.WARNING, "error while generating HTML", t);
+    }
+  }
+
+  /**
+   * Apply updated trace configuration through the tracerProvider based on query parameters.
+   *
+   * @param queryMap the map containing URL query parameters.
+   */
+  private void applyTraceConfig(Map<String, String> queryMap) {
+    String action = queryMap.get(QUERY_STRING_ACTION);
+    if (action == null) {
+      return;
+    }
+    if (action.equals(QUERY_STRING_ACTION_CHANGE)) {
+      TraceConfig.Builder newConfigBuilder = this.tracerProvider.getActiveTraceConfig().toBuilder();
+      String samplingProbabilityStr = queryMap.get(QUERY_STRING_SAMPLING_PROBABILITY);
+      if (samplingProbabilityStr != null && !samplingProbabilityStr.isEmpty()) {
```

Comment:
I think we can add `omitEmptyValues` to the trimmer in `parseQueryMap` to avoid having to check here

Reply:
With `omitEmptyValues()`, `isEmpty()` check can be skipped, but not null check right?

Reply:
Yup sorry just meant the empty check